packages/lesswrong/server/languageModels/languageModelIntegration.ts

```diff
@@ -0,0 +1,307 @@
+
+const openAIApiKey = new DatabaseServerSetting<string|null>('languageModels.openai.apiKey', null);
+const openAIOrganizationId = new DatabaseServerSetting<string|null>('languageModels.openai.organizationId', null);
+
+let openAIApi: OpenAIApi|null = null;
+export async function getOpenAI(): Promise<OpenAIApi|null> {
+  if (!openAIApi){
+    const apiKey = openAIApiKey.get();
+    const organizationId = openAIOrganizationId.get();
+    
+    if (apiKey) {
+      openAIApi = new OpenAIApi(new OpenAIApiConfiguration({
+        apiKey,
+        organization: organizationId ?? undefined,
+      }));
+    }
+  }
+  return openAIApi;
+}
+
+type LanguageModelAPI = "disabled"|"stub"|"openai";
+type LanguageModelClassificationTask = "isSpam"|"isFrontpage";
+type LanguageModelGenerationTask = "summarize"|"authorFeedback";
+type LanguageModelTask = LanguageModelClassificationTask|LanguageModelGenerationTask;
+
+export type LanguageModelTemplate = {
+  header: Record<string,string>
+  body: string
+}
+
+type LanguageModelConfig = {
+  template: LanguageModelTemplate,
+  task: LanguageModelTask,
+  api: LanguageModelAPI,
+  model: string,
+}
+
+type LanguageModelJob = LanguageModelConfig & {
+  maxTokens: number
+  inputs: Record<string,string>
+}
+
+/**
+ * canDoLanguageModelTask: Returns true if we're configured with an API key,
+ * prompts and anything else necessary to perform a language-model-based task.
+ * If this is false, UI for corresponding language-model-based features should
+ * not be displayed on the client.
+ *
+ * (Attempting to run language-model tasks may still fail even if this returned
+ * true, eg if we get an error from the OpenAI API).
+ */
+export async function canDoLanguageModelTask(task: LanguageModelTask, context: ResolverContext): Promise<boolean> {
+  const config = await getLMConfigForTask(task, context);
+  return (config && config.api!=="disabled");
+}
+
+function taskToWikiSlug(task: LanguageModelTask): string {
+  return `lm-config-${task.toLowerCase()}`;
+}
+
+/**
+ * Given a language-model task type, retrieve a corresponding template. Uses the
+ * text of an admin-only wiki page, formatted as Markdown.
+ */
+async function getLMConfigForTask(task: LanguageModelTask, context: ResolverContext): Promise<LanguageModelConfig> {
+  const wikiPageSlug = taskToWikiSlug(task);
+  const tag = await Tags.findOne({slug: wikiPageSlug});
+  
+  if (tag) {
+    return tagToLMConfig(tag, task);
+  } else {
+    return {
+      task,
+      api: "stub",
+      template: {
+        header: {api: "stub"},
+        body: "Test ${input}", //eslint-disable-line no-template-curly-in-string
+      },
+      model: "stub-model",
+    };
+  }
+}
+
+function tagToLMConfig(tag: DbTag, task: LanguageModelTask): LanguageModelConfig {
+  if (!tag) throw new Error("Tag not found");
+  
+  const template = wikiPageToTemplate(tag);
+  const {header, body} = template;
+  return {
+    api: (header["api"] ?? "disabled") as LanguageModelAPI,
```

Comment:
What guarantees that this is going to fit `LanguageModelAPI`?  It looks like we're parsing it out of the `DbTag` contents directly in `wikiPageToTemplate`